### DISCUSSION TITLE
Dummy databas interface

### DISCUSSION THREAD

**Discussion by Ojani** [uprm-inso4101-2024-2025-s1/semester-project-campus-resource-and-event-management#2] [Ideas]
## Problem
We will obviously need to store data for this project, however, setting up a database initially might stall out the development process early on. At the same time, I think there should be an interface made to abstract database operations, in other words, having a set of helper functions to perform different operations on the database instead of interacting with it directly. 

## Proposal
I propose that to make the development process simpler, these interfaces should be made on demand (on a dedicated file) as we need to access and modify different data, but instead of interacting with an actual database, it could interact with dummy data (perhaps a simple JSON file) so that the functionality that makes use of the data can be made. Later on, the interfaces can be made to retrieve and modify the data from the database instead of a JSON file without having to change any of the code that relies on these CRUD operations since the interface won't change.

**Comment by JeanSanchezFelix** [comment]
Great idea! In the later milestone we talk about implementing the actual database
